### PR TITLE
Add signal handler for catching Ctrl-C on hack/e2e

### LINKED ISSUE #43051: e2e tests can't be interrupted by Ctrl-C
Once e2e tests are running, the tests keep running even if I press `Ctrl-C` and exit the process that controls the TTY.

But the tests actually keep running in the background and they keep printing stuff on a shell prompt which has already returned and thereby clobbering the STDOUT

### PR DESCRIPTION
**Pull request by oomichi**
**What this PR does / why we need it**:

When operating e2e test, hack/e2e.go process creates kubetest process.
To kill the kubetest process when stop e2e test with Ctrl-C, we need
to send the signal to the process because it also creates another
process and it needs to kill it.
This PR adds the signal handler on hack/e2e.go to kill the kubetest
process.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

fixes #43051

**Special notes for your reviewer**:

https://github.com/kubernetes/test-infra/pull/4154 is the part of kubetest.

**Release note**:

`NONE`

### CODE REVIEW

File: hack/e2e.go [hunk 1]
```diff
@@ -81,12 +82,21 @@ func main() {
...
 	c := exec.Command(cmd, args...)
 	c.Stdout = os.Stdout
 	c.Stderr = os.Stderr
 	if err := c.Start(); err != nil {
 		return err
 	}
+	go func() {
+		sig := <-sig_channel
+		if err := c.Process.Signal(sig); err != nil {
+			log.Fatalf("err: %v", err)
```
Comment (fejta) [inline hack/e2e.go]: Please make the error message easier for a human to understand the context, something like
`Fatalf("could not send %s signal %s: %v", cmd, sig, err)`

**Review by fejta — COMMENTED** [review]
Very cool! One quick comment

File: hack/e2e.go (hunk 1 above)
Comment (oomichi) [inline hack/e2e.go]: Thanks for your review.
Yeah, that should be. I will fix it soon.

**Comment by dims** [comment]
Go @oomichi !!!

**Comment by oomichi** [comment]
@dims hehe, I am not sure which GO you mean. I am not good at GOlang yet anyways ;-)

**Pull request merged**